ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+                        camp->FieldDescription[0]))
+            break;
+    }
+    if (i_camp == bd_xp->nFields)
+        return retorn;
+
+    if (retorn == 1)
+    {
+        if (strlen(camp->FieldDescription[0]) >
+            MM_MAX_LON_DESCRIPCIO_CAMP_DBF - 4 - n_digits_i)
+            camp->FieldDescription[0][mida_nom - 4 - n_digits_i] = '\0';
+        //if (camp->FieldDescription[0] + strlen(camp->FieldDescription[0]))
+        sprintf(camp->FieldDescription[0] + strlen(camp->FieldDescription[0]),
+                " (%u)", i);
+        for (i_camp = 0; i_camp < bd_xp->nFields; i_camp++)
+        {
+            if (bd_xp->pField + i_camp == camp)
+                continue;
+            if (!strcasecmp(bd_xp->pField[i_camp].FieldDescription[0],
+                            camp->FieldDescription[0]))
+                break;
+        }
+        if (i_camp == bd_xp->nFields)
+            return retorn;
+    }
+
+    retorn = 1;
+    if (strlen(camp->FieldDescription[0]) >
+        MM_MAX_LON_DESCRIPCIO_CAMP_DBF - 4 - n_digits_i)
+        camp->FieldDescription[0][mida_nom - 4 - n_digits_i] = '\0';
+    camp->FieldDescription[0][strlen(camp->FieldDescription[0]) - 4 -
+                              n_digits_i + 1] = '\0';
+    if (strlen(camp->FieldDescription[0]) > MM_MAX_LON_DESCRIPCIO_CAMP_DBF - 7)
+        camp->FieldDescription[0][mida_nom - 7] = '\0';
+    for (i++; i < (size_t)256; i++)
+    {
+        //if (camp->FieldDescription[0] + strlen(camp->FieldDescription[0]))
+        sprintf(camp->FieldDescription[0] + strlen(camp->FieldDescription[0]),
+                " (%u)", i);
+        for (i_camp = 0; i_camp < bd_xp->nFields; i_camp++)
+        {
+            if (bd_xp->pField + i_camp == camp)
+                continue;
+            if (!strcasecmp(bd_xp->pField[i_camp].FieldName, camp->FieldName))
+                break;
+        }
+        if (i_camp == bd_xp->nFields)
+            return retorn;
+    }
+    return 2;
+}  // End of MM_ModifyFieldNameAndDescriptorIfPresentBD_XP()
+
+static int MM_DuplicateMultilingualString(
+    char *(cadena_final[MM_NUM_IDIOMES_MD_MULTIDIOMA]),
+    const char *const(cadena_inicial[MM_NUM_IDIOMES_MD_MULTIDIOMA]))
+{
+    size_t i;
+
+    for (i = 0; i < MM_NUM_IDIOMES_MD_MULTIDIOMA; i++)
+    {
+        if (cadena_inicial[i])
+        {
+            if (nullptr == (cadena_final[i] = strdup(cadena_inicial[i])))
+                return 1;
+        }
+        else
+            cadena_final[i] = nullptr;
+    }
+    return 0;
+}
+
+int MM_DuplicateFieldDBXP(struct MM_FIELD *camp_final,
+                          const struct MM_FIELD *camp_inicial)
+{
+    *camp_final = *camp_inicial;
+
+    if (0 != MM_DuplicateMultilingualString(
+                 camp_final->Separator,
+                 (const char *const(*))camp_inicial->Separator))
+        return 1;
+
+    return 0;
+}
+
+char *MM_strnzcpy(char *dest, const char *src, size_t maxlen)
+{
+    size_t i = 0;
+    if (!src || maxlen == 0)
```

Comment:
Nitpicking, but if maxlen == 0, we shouldn't write anything into dest.

Reply:
True.I adopt that:
if (maxlen == 0)
        return dest;
    
    if (!src)
    {
        *dest = '\0';
        return dest;
    }